### PR TITLE
Adding script for getting the preflight results

### PR DESCRIPTION
**Pull request by MarcinGinszt**
Related PR: https://github.com/redhat-openshift-ecosystem/operator-pipelines/pull/21
Jira: ISV-910

### CODE REVIEW

File: operatorcert/__init__.py [hunk 1]
```diff
@@ -311,3 +312,59 @@ def verify_pr_uniqueness(
...
+
+    # If there are multiple results, we only want the most recent one- we enforce it by sorting by creation_date
+    # and getting only the first result.
+    test_results_url = urljoin(
+        args.pyxis_url,
+        f"v1/projects/certification/id/{args.cert_project_id}/{resource}?"
+        f"filter=certification_hash=='{args.certification_hash}';"
+        f"version=='{args.operator_package_version}';"
+        f"operator_package_name=='{args.operator_name}'"
+        f"&sort_by=creation_date[desc]&page_size=1",
```
Comment (Allda) [inline operatorcert/__init__.py:336]: There can be more than just one artifact produced by a single pre-flight test. There should be at least a test log artifact + additional files produced by tests.

File: operatorcert/__init__.py [hunk 2]
```diff
@@ -311,3 +312,59 @@ def verify_pr_uniqueness(
...
+    )
+
+    rsp = requests.get(
+        test_results_url, cert=(args.cert_path, args.key_path), verify=False
```
Comment (Allda) [inline operatorcert/__init__.py:340]: verify=True

File: operatorcert/__init__.py [hunk 3]
```diff
@@ -311,3 +312,59 @@ def verify_pr_uniqueness(
...
+
+    if len(query_results) == 0:
+        logging.error(f"There is no {resource} for given parameters")
+        return None
+
+    # Get needed data from the query result
+    # artifacts
+    if resource == "artifacts":
+        result = base64.b64decode(query_results[0]["content"]).decode("utf-8")
+        file_path = "artifact.txt"
```
Comment (Allda) [inline operatorcert/__init__.py:353]: The file name can be gathered from the pyxis artifact response.

Comment (MarcinGinszt) [reply operatorcert/__init__.py:353]: I'm using the fixed file name to pass the script output to tekton step

File: operatorcert/__init__.py (hunk 1 above)
Comment (MarcinGinszt) [reply operatorcert/__init__.py:336]: So.. Do we want all of the artifacts for the same cert_project, cert_hash, version and operator package name?

File: tests/test_utils.py [hunk 4]
```diff
@@ -30,14 +30,18 @@ def test_find_file(tmp_path: Path) -> None:
 
 
 def test_store_results() -> None:
-    results = {"example_name": "example_value", "other_name": "other_value"}
+    results = {
+        "example_name": "example_value",
+        "other_name": "other_value",
+        "none": None,
```
Comment (wcheang) [inline tests/test_utils.py:36]: Does this actually work? I just tried on my python3 terminal, and file.write(None) throws an exception.

File: operatorcert/__init__.py [hunk 5]
```diff
@@ -311,3 +312,59 @@ def verify_pr_uniqueness(
...
+        raise ValueError(
+            f"Preflight results are either test-results or artifacts. Got {resource} instead"
+        )
+
+    # If there are multiple results, we only want the most recent one- we enforce it by sorting by creation_date
+    # and getting only the first result.
+    test_results_url = urljoin(
+        args.pyxis_url,
+        f"v1/projects/certification/id/{args.cert_project_id}/{resource}?"
+        f"filter=certification_hash=='{args.certification_hash}';"
```
Comment (wcheang) [inline operatorcert/__init__.py:333]: Is the hash gonna need some sort of escaping to be a valid url?

Comment (MarcinGinszt) [reply operatorcert/__init__.py:332]: it works in it's current form :slightly_smiling_face:

File: tests/test_utils.py (hunk 4 above)
Comment (MarcinGinszt) [reply tests/test_utils.py:36]: Good point, but that's what we test here! In the store_results function body: 
```
        if result_value is None:
            result_value = ""
```

File: operatorcert/entrypoints/download_artifacts.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+            # resource not found
+            break
+
+    # Store results
+    store_results(
+        {
+            "results_exists": str(
+                None not in resource_ids.values()
+            ),  # 'False' if at least one id is set to None
+            "test_result_id": resource_ids.get("test-results"),
```
[the remark this answers is not in the document]
Comment (MarcinGinszt) [reply operatorcert/entrypoints/download_artifacts.py]: I don't really like the idea to add the arguments for the sake of informing users. I've added this information to the docstring. WDYT?

File: .gitignore [hunk 7]
```diff
@@ -131,4 +139,4 @@ dmypy.json
 .idea/
 
 # Vim swap files
-.*.sw?
```
Comment (Allda) [inline .gitignore:134]: Missing new line at the end of the file.

**Review by Allda — APPROVED** [review]

**Pull request merged**